service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalogHandler.java

```diff
@@ -997,7 +997,7 @@ public void commitTransaction(CommitTransactionRequest commitTransactionRequest)
             callContext.getPolarisCallContext(), pendingUpdates);
     if (!result.isSuccess()) {
       // TODO: Retries and server-side cleanup on failure
-      throw new CommitFailedException(
+      throw new CommitConflictException(
           "Transaction commit failed with status: %s, extraInfo: %s",
           result.getReturnStatus(), result.getExtraInformation());
```

Comment:
Strictly speaking the majority of possible status codes do not fit 409 responses, from my POV. It might be preferable to handle this case by inspecting `result.getReturnStatus()` (probably in the exception mapper).

That said, such a change will break current de-facto API contracts.... I'm not sure whether there is appetite in the community for clarifying this.